src/tools/genpolicy/src/yaml.rs

```diff
@@ -442,3 +443,18 @@ pub fn get_sysctls(security_context: &Option<pod::PodSecurityContext>) -> Vec<po
     }
     vec![]
 }
+
+/// Constructs a non-anchored regex for an object according to k8s naming conventions:
+/// 1. If the name field is set, return that literally.
+/// 2. If name is unset but generateName is set, return regex that matches generateName and a random suffix.
+/// 3. Otherwise, return None. This object is not considered valid by the k8s API server!
+pub fn name_regex_from_meta(meta: &ObjectMeta) -> Option<String> {
+    let generateName = meta
```

Comment:
Here we could do 
```
meta.name.clone().or_else(|| {
        meta.generateName
            .as_ref()
            .map(|p| format!("{}{}", regex::escape(p), GENERATE_NAME_SUFFIX_REGEX))
    })
```

In order to only build `generateName` if `name` is absent